### PR TITLE
add possibility to use certificate in the cart

### PR DESCRIPTION
**Pull request by Tolik170**
## Description

Add possibility to use certificate in the cart.

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/hooks/use-cart.js [hunk 1]
```diff
@@ -36,21 +36,39 @@ export const useCart = (user = null) => {
...
+        return [getPriceWithCurrency(Math.round(price - (price / 100) * discount)), item.quantity];
+        
       }
-      return [Math.round(price), item.quantity];
+      return [getPriceWithCurrency(price), item.quantity];
     });
     return newArr.reduce(
-      (acc, item) => acc + getPriceWithCurrency(calcPriceForCart(item[0], item[1])),
+      (acc, item) => {
+         return acc + calcPriceForCart(item[0], item[1])},
```
Comment (Pipuss) [inline src/hooks/use-cart.js:51]: I don't think its intuitively clear what item[0],..etc mean. Maybe it make sence to destructure them first?

File: src/hooks/use-cart.js [hunk 2]
```diff
@@ -36,21 +36,39 @@ export const useCart = (user = null) => {
...
+
+    const newArr = cart.map((item) => {
+      const price = getPriceWithCurrency(item.sizeAndPrice.price);
+      const priceWithCertificate = price;
+      return [priceWithCertificate - value, item.quantity, price];
+    });
+
+    return newArr.reduce( 
+      (acc, item) => {
+        return acc + calcPriceWithCertificateForCart(item[0], item[1], item[2])},
```
Comment (Pipuss) [inline src/hooks/use-cart.js:67]: And here

**Review by Pipuss — APPROVED** [review]

File: src/containers/orders/cart/filled-cart/tests/filled-cart.variables.js [hunk 3]
```diff
@@ -45,6 +46,32 @@ export const mockPromoCode = {
...
+      getCertificateByName: {
+        dateEnd: '2023-07-20T15:06:30.256Z',
+        dateStart: '2022-07-19T15:06:30.256Z',
+        isActivated: true,
+        isExpired: false,
+        isUsed: false,
+        name: 'HOR40315176',
+        value: 17,
+        _id: '62d6cab33cb3cc2908fc7d61',
+        __typename: 'PaginatedCertificate'
```
Comment (Pipuss) [inline src/containers/orders/cart/filled-cart/tests/filled-cart.variables.js:69]: Weird naming with __

**Review by Pipuss — APPROVED** [review]

File: src/containers/orders/cart/filled-cart/tests/filled-cart.variables.js (hunk 3 above)
Comment (Tolik170) [reply src/containers/orders/cart/filled-cart/tests/filled-cart.variables.js:69]: __typename is a graph ql field which returns the object type's name.

**Review by maarekua — DISMISSED** [review]

**Review by MickhaelSh — APPROVED** [review]

**Review by greentoss — APPROVED** [review]

**Pull request merged**